### PR TITLE
feat: allow to upload larger images without cropping any part of the image

### PR DESCRIPTION
**Pull request by AfonsoMartins26**

### CODE REVIEW

**Comment by MarioRodrigues10** [comment]
First of all, i really like the improvements that you made, but let's review a few things.

1) I think it will be good to keep the same layout for vertical images, and use a similar layout to horizontal images.
image

2) The use case should be like, I upload a picture to be added to CeSIUMPeloMundo and you choose the picture orientation, by default is vertical, in case it's horizontal you display the layout that you showed us by using the same design Diogo did previously.

You can make that by adding a prop either to json and the function that parses the pictures.  

What's your thoughts on it? @AfonsoMartins26

**Comment by MarioRodrigues10** [comment]
It looks amazing @AfonsoMartins26  🙌  !!
Great job 🤩

File: components/Marker/Marker.tsx [hunk 1]
```diff
@@ -15,47 +14,49 @@ const Marker = ({
   country,
   author,
   photo,
-  date
+  date,
+  orientation = 'default' // Default to 'default' orientation
```
Comment (MarioRodrigues10) [inline components/Marker/Marker.tsx:18]:
```suggestion
  orientation = 'vertical'
```
Vertical should be the default orientation

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

**Review by MarioRodrigues10 — DISMISSED** [review]
This looks good to me ✅ , but let the other reviewers @diogogmatos , @joaodiaslobo and @ruilopesm take a look into it !! 🙌

**Review by joaodiaslobo — CHANGES_REQUESTED** [review]
The image now has some odd white borders on its corners, can you fix that?
Also, I think the location text looks less clear than before, missing a shadow maybe?

New:
image
Old:
image

**Review by joaodiaslobo — CHANGES_REQUESTED** [review]
Also, I wonder if it would be possible to maybe discover the orientation of the image automatically through its width and height? 🤔

**Review by joaodiaslobo — CHANGES_REQUESTED** [review]
Popover is way too small on mobile devices.

New | Old
image

**Comment by AfonsoMartins26** [comment]
> Also, I wonder if it would be possible to maybe discover the orientation of the image automatically through its width and height? 🤔

In my initial pr, I implemented the feature. However, @MarioRodrigues10 said:
"The use case should work as follows: When uploading a picture to be added to CeSIUMPeloMundo, you should be able to choose the picture orientation. The default orientation should be vertical. In cases where the picture is horizontal, you should display the layout you showed us earlier, using the same design that Diogo previously implemented."

**Comment by AfonsoMartins26** [comment]
Pc and mobile changes:
Screenshot from 2024-08-06 18-46-18
Screenshot from 2024-08-06 18-47-15

**Review by joaodiaslobo — CHANGES_REQUESTED** [review]
Text looks cool and no more white borders, great! 🎉
But now, the image is way too big on mobile, to the point where it gets out of the screen:
image
image

How it should look:
image

I have used iPhone SE resolution for these examples.
Just make sure it fits and this PR should be good to go!! 🙏

**Review by ruilopesm — COMMENTED** [review]
It would be very nice and clever trying to guess the orientation of the photo fully based on its dimensions 👀

**Comment by MarioRodrigues10** [comment]
@ruilopesm  I had the same thought initially, but I changed my mind when thinking about some edge cases.
First of all in case where is a squared image, how will it would be defined by default?

Now let's imagine I have a horizontal picture but I want only to show part of it as a vertical image, the automatically display would put it by default has a horizontal picture.

I would like to give the user the ability to choose which orientation of the picture he wants.

**Comment by ruilopesm** [comment]
@MarioRodrigues10 I think we can guess the best orientation by the image's aspect ratio. If perfect, let it be vertical. As for the other part, about the user deciding which orientation they want - we can let the user decide it, and if no orientation is defined, we automatically _calculate_ one. Nonetheless, we can also have an opinion about the best orientation when reviewing future PRs. What do you think? 👀

**Comment by MarioRodrigues10** [comment]
@ruilopesm it seems good, so let's resume it, basically if the user doesn't pass the orientation as prop it will automatically decide which one fits better. 
If the user decides to pass the orientation as prop we use the one the user has decided.

File: styles/globals.css
Comment (MarioRodrigues10) [inline styles/globals.css:1]: Add this to remove the white border
```css
.leaflet-popup-content-wrapper, .leaflet-popup-tip {
  background: transparent !important;
  border: none !important;
  border-radius: 12px;
}
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

File: components/Marker/Marker.tsx [hunk 2]
```diff
@@ -15,47 +15,73 @@ const Marker = ({
...
+        console.error('Failed to load image:', photo);
+        setImageOrientation('vertical');
+      };
+    }
+  }, [photo, orientation]);
+
+  const popupClassName =
+    imageOrientation === 'horizontal' ? '651:400' : '301:470';
+
+  const [width, height] = popupClassName.split(':').map(Number);
```
Comment (MarioRodrigues10) [inline components/Marker/Marker.tsx:49]:
```suggestion
  const handleImageLoad = (img: HTMLImageElement): void => {
    setImageOrientation(img.width > img.height ? 'horizontal' : 'vertical');
  }

  useEffect(() => {
    if(orientation || !photo) return;

    const img = new window.Image();
    img.src = photo;

    img.onload = () => handleImageLoad(img);
  }, [photo, orientation]);

  const popupClassName = useMemo(() => imageOrientation === 'horizontal' ? '651:400' : '301:470', [imageOrientation]);

  const [width, height] = useMemo(() => popupClassName.split(':').map(Number), [popupClassName]);
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

**Review by joaodiaslobo — CHANGES_REQUESTED** [review]
On mobile, the images are **always** vertical, even when the `orientation` is explicitly set as `horizontal` on the data. Is this behavior intended? 🤔

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]
image
Overlay text needs to be adjust to the mobile perspective. 🙌

**Comment by diogogmatos** [comment]
image

Long text is looking weird when it wraps around. Don't know if this was an existing problem or a new problem though ahaha.

**Review by diogogmatos — COMMENTED** [review]
image

In some pins the image shows up sideways even though the picture orientation is correct in the picture file. This is something that already happens in the deployed version but it would be cool to fix it too since we're messing with this.

File: components/Marker/Marker.tsx [hunk 3]
```diff
@@ -15,47 +15,76 @@ const Marker = ({
...
+
+  const popupClassName = useMemo(() => {
+    const isMobile = window.innerWidth <= 500;
+
+    if (isMobile) {
+      return imageOrientation === 'horizontal' ? '401:250' : '351:550';
+    } else {
+      return imageOrientation === 'horizontal' ? '651:400' : '301:470';
+    }
+  }, [imageOrientation]);
```
Comment (MarioRodrigues10) [inline components/Marker/Marker.tsx:40]:
```suggestion
  const popupClassName = useMemo(() => {
    const isMobile = window.innerWidth <= 500;

    const sizeMapping = isMobile ? mobileSizes : desktopSizes;
    return imageOrientation === 'horizontal' ? sizeMapping.horizontal : sizeMapping.vertical;
  }, [imageOrientation]);
```

File: components/Marker/Marker.tsx [hunk 4]
```diff
@@ -15,47 +15,76 @@ const Marker = ({
...
   author,
   photo,
-  date
+  date,
+  orientation
 }: IPin) => {
-  const icon = getIcon(type);
-
+  const icon = useMemo(() => getIcon(type), [type]);
   const name = useMemo(() => getNameString(author), [author]);
```
Comment (MarioRodrigues10) [inline components/Marker/Marker.tsx:23]:
```suggestion
    const mobileSizes = {
      horizontal: '401:250',
      vertical: '351:550',
    };

    const desktopSizes = {
      horizontal: '651:400',
      vertical: '301:470',
    };
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

**Comment by joaodiaslobo** [comment]
> In some pins the image shows up sideways even though the picture orientation is correct in the picture file. This is something that already happens in the deployed version but it would be cool to fix it too since we're messing with this.


@diogogmatos 
Pretty sure this is already fixed in this PR!

**Comment by diogogmatos** [comment]
> > In some pins the image shows up sideways even though the picture orientation is correct in the picture file. This is something that already happens in the deployed version but it would be cool to fix it too since we're messing with this.
> 
> 
> 
> 
> 
> @diogogmatos 
> 
> Pretty sure this is already fixed in this PR!
> 
> 

The screenshot was from the deploy preview on this PR...

**Comment by joaodiaslobo** [comment]
> The screenshot was from the deploy preview on this PR...

@diogogmatos 
Interesting, locally it does in fact use the right orientation but in the deploy preview it doesn't. 🤔

**Review by diogogmatos — COMMENTED** [review]
Seems to be fixed! Just make sure the checks pass ;)

**Comment by MarioRodrigues10** [comment]
Yes problem seems to be fixed @diogogmatos, I think @joaodiaslobo requested him to remove one or two lines and fix the CI, and after that it's good to go. :raised_hands:

**Review by joaodiaslobo — APPROVED** [review]
What a great addition! 🤩

**Pull request merged**